tower-grpc-interop/src/client.rs

```diff
@@ -700,6 +700,78 @@ impl TestClients {
                 future::ok::<Vec<TestAssertion>, Box<Error>>(assertions)
             })
     }
+
+    fn custom_metadata_test(
+        &mut self,
+    ) -> impl Future<Item = Vec<TestAssertion>, Error = Box<Error>> {
+        let key1 = "x-grpc-test-echo-initial";
+        let value1 = MetadataValue::from_str("test_initial_metadata_value").unwrap();
+        let key2 = "x-grpc-test-echo-trailing-bin";
+        let value2 = MetadataValue::from_bytes(&[0xab, 0xab, 0xab]);
+
+        let req = SimpleRequest {
+            response_type: pb::PayloadType::Compressable as i32,
+            response_size: LARGE_RSP_SIZE,
+            payload: Some(util::client_payload(LARGE_REQ_SIZE)),
+            ..Default::default()
+        };
+        let mut req_unary = Request::new(req);
+        req_unary.metadata_mut().insert(key1, value1.clone());
+        req_unary.metadata_mut().insert_bin(key2, value2.clone());
+
+        let stream = stream::iter_ok(vec![make_ping_pong_request(0)]);
+        let mut req_stream = Request::new(stream);
+        req_unary.metadata_mut().insert(key1, value1.clone());
+        req_unary.metadata_mut().insert_bin(key2, value2.clone());
+
+        let value1_cloned = value1.clone();
+        let value2_cloned = value2.clone();
+        let unary = self
+            .test_client
+            .unary_call(req_unary)
+            .map(move |e| {
+                vec![
+                    test_assert!(
+                        "metadata string must match in unary",
+                        e.metadata().get(key1) == Some(&value1_cloned),
+                        format!("result={:?}", e.metadata().get(key1))
+                    ),
+                    // TODO trailing metadata not implemented?
```

Comment:
I'm a little bit confused about what you mean with this comment. Is this a TODO that states that this is broken and that it should be fixed (implying that the test is not fully completed yet)? If so I think it would be nice to say that explicitly, and perhaps point to issue #27 to provide a little bit of context.

If it is broken, is it broken when run against the Go interop server (implying client side issue) or only when run against the tower-grpc server?